In [1]:
import pandas as pd
import statsmodels.api as sm
from termcolor import colored as cl

In [2]:
data = pd.read_csv('projects.csv')


x_var = data[['Estimate']] # independent variable
y_var = data['Final'] # dependent variable

Fit and summary

In [3]:
slr_model = sm.OLS(y_var, x_var) # Ordinary Least Squares 
slr_reg = slr_model.fit()
print(cl(slr_reg.summary(),attrs = ['bold']))

                                 OLS Regression Results                                
Dep. Variable:                  Final   R-squared (uncentered):                   0.756
Model:                            OLS   Adj. R-squared (uncentered):              0.750
Method:                 Least Squares   F-statistic:                              111.7
Date:                Tue, 23 Nov 2021   Prob (F-statistic):                    1.38e-12
Time:                        13:28:09   Log-Likelihood:                         -134.71
No. Observations:                  37   AIC:                                      271.4
Df Residuals:                      36   BIC:                                      273.0
Df Model:                           1                                                  
Covariance Type:            nonrobust                                                  
                 coef    std err          t      P>|t|      [0.025      0.975]
-----------------------------------------

In [4]:
print("Parameters: ", slr_reg.params)
print("R2: ", slr_reg.rsquared)

Parameters:  Estimate    1.411537
dtype: float64
R2:  0.7563365048110292


This model has been trained based on the historical data that we fed it with. Based on the results, 75.6% variation in y (final cost) is explained by X (original estimate). 

Based on the OLS Regression Results, in order to calculate the cost for the future projects we should multiply the estimated deadline on the Estimate coef, which is **1.41**

To predict future 'Final' cost of the project, we create a function.

In [5]:
def prediction(estimate):
    coefficient = slr_reg.params[0]
    formula = coefficient*estimate
    return formula

In [6]:
prediction(12)

16.938442723196083